### PR TITLE
frontend/refactor/rentals: #4998 CompareUTCDate refactored, polling refactored 

### PR DESCRIPTION
**Pull request by slow-codex**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description

- Changed compareUTCDate from per milliseconds to per second difference between two UTC, and return type from Effect String to Int.
- Refactored the rentalAndIntercity OnConfirm Polling with the current polling for Confirming Quotes.
- Fixed RideScheduled Banner being sticky even if the local stage was changing.
- Handled minimized app case while polling, and after reopening the app from minimized the loader will be in the same exact state.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 1]
```diff
@@ -176,10 +176,14 @@ screen initialState =
                   void $ pure $ setValueToLocalStore TRACKING_ID (getNewTrackingId unit)
                   let pollingCount = ceil ((toNumber initialState.props.searchExpire)/((fromMaybe 0.0 (NUM.fromString (getValueToLocalStore TEST_POLLING_INTERVAL))) / 1000.0))
                   void $ launchAff $ flowRunner defaultGlobalState $ getQuotesPolling (getValueToLocalStore TRACKING_ID) GetQuotesList Restart pollingCount (fromMaybe 0.0 (NUM.fromString (getValueToLocalStore TEST_POLLING_INTERVAL))) push initialState
-              ConfirmingRide -> void $ launchAff $ flowRunner defaultGlobalState $ confirmRide GetRideConfirmation 5 3000.0 push initialState
-                
-              ConfirmingQuotes -> void $ launchAff $ flowRunner defaultGlobalState $ rentalAndIntercityConfirmRide GetRideConfirmation CheckFlowStatusAction 100 3000.0 push initialState
-
+              ConfirmingRide -> void $ launchAff $ flowRunner defaultGlobalState $ confirmRide GetRideConfirmation CheckFlowStatusAction 5 3000.0 push initialState
+              ConfirmingQuotes -> do 
+                let _ = spy "dsafasdadsfsadf" (getValueToLocalStore CONFIRM_QUOTES_POLLING)
```
Comment (aryan-13) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:181]: Please remove logs from all the places

File: Frontend/ui-common/src/Engineering/Helpers/Commons.js [hunk 2]
```diff
@@ -547,7 +547,7 @@ export const getUTCAfterNSecondsImpl = function (str, seconds) {
 export const compareUTCDateImpl = function (date1, date2) {
   const d1 = new Date(date1);
   const d2 = new Date(date2);
-  return d1.getTime() - d2.getTime();
+  return Math.floor((d1.getTime() - d2.getTime())/1000);
```
Comment (aryan-13) [inline Frontend/ui-common/src/Engineering/Helpers/Commons.js:550]: Can we use `getDifferenceBetweenDates`  and remove this function

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 3]
```diff
@@ -2841,57 +2845,32 @@ driverLocationTracking push action driverArrivedAction updateState duration trac
...
         _ <- pure $ printLog "api Results " response
         let (RideBookingRes resp) = response
-            fareProductType = getFareProductType $ (resp.bookingDetails) ^. _fareProductType
-            status = if any ( _ == fareProductType) [ FPT.RENTAL, FPT.INTER_CITY ] then "TRIP_ASSIGNED" else "CONFIRMED"
-        if status == resp.status && not (null resp.rideList) then do
-            doAff do liftEffect $ push $ action response
-            pure unit
+            fareProductType = (resp.bookingDetails) ^. _fareProductType
+            status = if any ( _ == fareProductType ) ["OneWaySpecialZoneAPIDetails"] then "CONFIRMED" else "TRIP_ASSIGNED"
+            willRideListNull = any ( _ == fareProductType ) ["OneWaySpecialZoneAPIDetails"]
```
Comment (aryan-13) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:2859]: use Enums here

**Review by aryan-13 — CHANGES_REQUESTED** [review]

File: Frontend/ui-common/src/Engineering/Helpers/Commons.js (hunk 2 above)
Comment (slow-codex) [reply Frontend/ui-common/src/Engineering/Helpers/Commons.js:550]: `getDifferenceBetweenDates` is `ui-common` specific so I removed that function in `Utils.js` and used this only for both `ui-customer` and `ui-driver`.

**Review by aryan-13 — APPROVED** [review]

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 4]
```diff
@@ -2726,8 +2731,10 @@ eval (DateTimePickerAction dateResp year month day timeResp hour minute) state =
 
 eval (LocationTagBarAC (LocationTagBarV2Controller.TagClicked tag)) state = do 
   case tag of 
-    "RENTALS" -> exit $ GoToRentalsFlow state
-    "INTER_CITY" -> continue state{data{ source=(getString CURRENT_LOCATION)}, props{isSource = Just false, canScheduleRide = true, isSearchLocation = SearchLocation, currentStage = SearchLocationModel, searchLocationModelProps{crossBtnSrcVisibility = false, findPlaceIllustration = null state.data.locationList }}}
+    "RENTALS" -> exit $ GoToRentalsFlow state { data {rentalsInfo = Nothing } }
+    "INTER_CITY" -> do
+      _ <- pure $ updateLocalStage SearchLocationModel
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:2736]: use void

**Review by MercyQueen — APPROVED** [review]

**Pull request merged**